luma_core/src/ios.rs

```diff
@@ -0,0 +1,225 @@
+//! ``ios`` module of ``luma_core``.
+//!
+//! Contains functions for access to the Starlet when running IOS
+
+use crate::cache::DCFlushRange;
+use crate::io::{read32, write32};
+use alloc::boxed::Box;
+
+const BASE: u32 = 0x0d000000;
+
+const HW_IPC_PPCMSG: u32 = BASE + 0;
+const HW_IPC_PPCCTRL: u32 = BASE + 4;
+const HW_IPC_ARMMSG: u32 = BASE + 8;
+
+/// The type of a file descriptor in IOS.
+pub type RawFd = i32;
```

Comment:
The file descriptor is never less then zero, using a `u32` would be a better option